core/chat/events/events.go

```diff
@@ -66,6 +73,105 @@ func (e *UserMessageEvent) SetDefaults() {
 	e.RenderAndSanitizeMessageBody()
 }
 
+// implements the emojiDef.Emojis interface but uses case-insensitive search.
+// the .children field isn't currently used, but could be used in a future
+// implementation of say, emoji packs where a child represents a pack.
+type emojis struct {
+	list     []emojiDef.Emoji
+	names    map[string]*emojiDef.Emoji
+	children []emojiDef.Emojis
+}
+
+// return a new Emojis set.
+func newEmojis(emotes ...emojiDef.Emoji) emojiDef.Emojis {
+	self := &emojis{
+		list:     emotes,
+		names:    map[string]*emojiDef.Emoji{},
+		children: []emojiDef.Emojis{},
+	}
+
+	for i := range self.list {
+		emoji := &self.list[i]
+		for _, s := range emoji.ShortNames {
+			self.names[s] = emoji
+		}
+	}
+
+	return self
+}
+
+func (self *emojis) Get(shortName string) (*emojiDef.Emoji, bool) {
+	v, ok := self.names[strings.ToLower(shortName)]
+	if ok {
+		return v, ok
+	}
+
+	for _, child := range self.children {
+		v, ok := child.Get(shortName)
+		if ok {
+			return v, ok
+		}
+	}
+
+	return nil, false
+}
+
+func (self *emojis) Add(emotes emojiDef.Emojis) {
+	self.children = append(self.children, emotes)
+}
+
+func (self *emojis) Clone() emojiDef.Emojis {
+	clone := &emojis{
+		list:     self.list,
+		names:    self.names,
+		children: make([]emojiDef.Emojis, len(self.children)),
+	}
+
+	copy(clone.children, self.children)
+
+	return clone
+}
+
+var (
+	emojiMu           sync.Mutex
+	emojiDefs         = newEmojis()
+	emojiHTML         = make(map[string]string)
+	emojiModTime      time.Time
+	emojiHTMLFormat   = `<img src="{{ .URL }}" class="emoji" alt=":{{ .Name }}:" title=":{{ .Name }}:">`
+	emojiHTMLTemplate = template.Must(template.New("emojiHTML").Parse(emojiHTMLFormat))
+)
+
+func loadEmoji() {
+	modTime, err := data.UpdateEmojiList(false)
+	if err != nil {
```

Comment:
This probably wasn't handled great before, but if `os.Stat` returns an error it won't get to the error log in `UpdateEmojiList` and will silently fail, so maybe it should be logged here.